apps/web/src/app/(profile)/[username]/_components/dashboard/badges/_difficulty_badges.ts

```diff
@@ -0,0 +1,58 @@
+import { prisma } from '@repo/db';
+import type {AllBadgeObjs, BadgeFn} from "~/app/(profile)/[username]/_components/dashboard/_actions";
+import type {SolutionBadges} from "~/app/(profile)/[username]/_components/dashboard/badges/_shared_solutions_badges";
+import type {AotBadges} from "~/app/(profile)/[username]/_components/dashboard/badges/_advent_badges";
+export const DifficultyBadgeKeys = ['BEGINNER', 'EASY', 'MEDIUM', 'HARD', 'EXTREME'] as const;
+
+export type DifficultyBadges = typeof DifficultyBadgeKeys[number];
+export interface Difficulty {
+  Difficulty: DifficultyBadges;
+  TotalCompleted: number;
+}
+
+export const difficultyBadgesFn: BadgeFn =
+  async ({
+           userId,
+           badges,
+         }: {
+    userId: string;
+    badges: AllBadgeObjs;
+  }) => {
+    // DifficultyBadges
+    const difficulty: Difficulty[] | null = await DifficultyRetrieveData(userId);
+    return await DifficultyBadgesFn(badges, difficulty ?? []);
+  };
+
+export async function DifficultyRetrieveData(userId: string) {
+  const data: Difficulty[] | null =
+    await prisma.$queryRaw`SELECT Difficulty, COUNT(Id) as TotalCompleted FROM (SELECT DISTINCT Difficulty, Challenge.Id FROM Submission JOIN Challenge ON Submission.challengeId = Challenge.Id WHERE Submission.userId = ${userId} AND IsSuccessful = 1) unique_query GROUP BY Difficulty `;
+  return data;
+}
+
+export const AwardDifficultyBadge = (slug: AotBadges | DifficultyBadges | SolutionBadges ): slug is DifficultyBadges => {
```

Comment:
camelcase

Reply:
I've tried to fix the casing where I got it wrong. Not confident I got it all.

interfaces are PascalCase
types are PascalCase
functions are camelCase
objects are camelCase
const/let are camelCase

Correct me where I'm wrong.

Reply:
this is is the full casing reference:
interfaces are PascalCase
types are PascalCase
functions are camelCase
variables (objects, var, let ecc...) are camelCase
constants (const) can be either  camelCase or SCREAMING_SNAKE_CASE,

Note on constants:
usually for constant objects you do camelCase and for numbers/strings you do SCREAMING_SNAKE_CASE but this is really up to preference (eg. I like to do SCREAMING_SNAKE_CASE for all my constants )

Reply:
closing for now - I think I got everything for casing.